src/sentry/integrations/gitlab/webhooks.py

```diff
@@ -193,12 +193,19 @@ def dispatch(self, request: Request, *args, **kwargs) -> Response:
         return super().dispatch(request, *args, **kwargs)
 
     def post(self, request: Request) -> Response:
+        extra = {
+            # This tells us the Gitlab version being used (e.g. current gitlab.com version -> GitLab/15.4.0-pre)
+            "user-agent": request.META.get("HTTP_USER_AGENT"),
+            "event-type": request.META["HTTP_X_GITLAB_EVENT"],
+        }
         token = "<unknown>"
         try:
             # Munge the token to extract the integration external_id.
             # gitlab hook payloads don't give us enough unique context
             # to find data on our side so we embed one in the token.
             token = request.META["HTTP_X_GITLAB_TOKEN"]
+            # This is an intentional circumvention of data scrubbing in order to debug customer issues
+            extra["tooken_from_gitlab"] = token
```

Comment:
You can see this in [this issue](https://sentry.io/organizations/sentry-test/issues/3612168558/?project=6730869).

It allows for visual inspection for customer support to validate what their webhook is sending us (right by the stack trace):
<img width="322" alt="image" src="https://user-images.githubusercontent.com/44410/191783609-7793de19-b7d9-4af2-bd20-880a556062fa.png">

Also available in the breadcrumbs:
<img width="391" alt="image" src="https://user-images.githubusercontent.com/44410/191783495-891ee41c-c058-4105-99a7-1099ef0b7b4d.png">

Reply:
Since we store IPs from our customers, I assuming this bit of information is okay to store.